paasta_tools/tron_tools.py

```diff
@@ -205,32 +207,47 @@ def get_cmd(self):
             command = "unset MESOS_DIRECTORY MESOS_SANDBOX; " + command
         return command
 
+    def get_spark_cluster(self):
+        return self.config_dict.get("spark_cluster", self.get_cluster())
+
+    def get_spark_pool(self):
+        return self.config_dict.get("spark_pool", "default")
+
     def get_env(self):
         env = super().get_env()
         spark_env = {}
         if self.get_executor() == "spark":
+            spark_ui_port = pick_random_port(
+                f"{self.get_service()}{self.get_instance()}".encode()
+            )
             spark_env = get_mesos_spark_env(
-                spark_app_name="tron_spark_{self.get_service()}_{self.get_instance()}",
-                spark_ui_port=pick_random_port(
-                    f"{self.get_service()}{self.get_instance()}".encode()
-                ),
-                mesos_leader=find_mesos_leader(self.get_cluster()),
+                spark_app_name=f"tron_spark_{self.get_service()}_{self.get_instance()}",
+                spark_ui_port=spark_ui_port,
+                mesos_leader=find_mesos_leader(self.get_spark_cluster()),
                 mesos_secret=load_mesos_secret_for_spark(),
-                paasta_cluster=self.get_cluster(),
-                paasta_pool=self.get_pool(),
+                paasta_cluster=self.get_spark_cluster(),
+                paasta_pool=self.get_spark_pool(),
                 paasta_service=self.get_service(),
                 paasta_instance=self.get_instance(),
                 docker_img=self.get_docker_url(),
-                volumes=format_volumes(
-                    self.get_volumes(load_system_paasta_config().get_volumes())
-                ),
+                volumes=[
+                    f"{v['hostPath']}:{v['containerPath']}:{v['mode']}"
+                    for v in self.get_volumes(load_system_paasta_config().get_volumes())
+                ],
                 user_spark_opts=self.config_dict.get("spark_args"),
                 event_log_dir=get_default_event_log_dir(
                     service=self.get_service(),
                     aws_credentials_yaml=self.config_dict.get("aws_credentials"),
                 ),
             )
+            env["EXECUTOR_CLUSTER"] = self.get_spark_cluster()
+            env["EXECUTOR_POOL"] = self.get_spark_pool()
```

Comment:
in retrospect I feel like we should have named these something like `CLUSTERMAN_CLUSTER` and `CLUSTERMAN_POOL` -- this is very spark-specific.

Reply:
well originally it was `SPARK_CLUSTER` and `SPARK_POOL`, and I changed them to `EXECUTOR` -- my reasoning here is that even if it's something not Spark, these env vars are dictating where the "executors" run, and that most things are going to have some sort of executor-ish concept.